pyBumpHunter/bumphunter_2dim.py

```diff
@@ -434,11 +450,280 @@ def _scan_hist(self, hist, ref, w_ar, ih: int):
+
+            min_Pval :
+                Minimum p_value obtained durring the scan (float).
+
+            min_loc :
+                Position of the window corresponding to the minimum p-value ([integer,integer]).
+
+            min_width :
+                Width of the window corresponding to the minimum p-value ([integer,integer]).
+
+            norm_scale :
+                The scale factor computed with side-band normalization (float).
+                If side-band normalization is not use, norm_scale is set to None.
+        """
+
+        # Initialize the global results for all channels
+        min_Pval_all = np.full(len(hist), 1.0, dtype=float)
+        min_loc_all = [[0, 0] for ch in range(len(hist))]
+        min_width_all = [[h.shape[0], h.shape[1]] for h in hist]
+        signal_eval_all = np.full(len(hist), 0.0)
+
+        # Compute the total number of event for sideband normalization
+        if self.use_sideband:
+            ref_total = []
+            hist_total = []
+            for ch in range(len(hist)):
+                ref_total.append(ref[ch].sum())
+                hist_total.append(hist[ch].sum())
+            min_scale_all = np.empty(len(hist))
+
+        # Compute scan_steppx for all width
+        if self.scan_step[0] == "full":
+            scan_steppx = [w[0] for w in w_ar]
+        elif self.scan_step[0] == "half":
+            scan_steppx = [max(1, w[0] // 2) for w in w_ar]
+        else:
+            scan_steppx = [self.scan_step[0] for w in w_ar]
+
+        # Compute scan_steppy for all width
+        if self.scan_step[1] == "full":
+            scan_steppy = [w[1] for w in w_ar]
+        elif self.scan_step[1] == "half":
+            scan_steppy[1] = [max(1, w[1] // 2) for w in w_ar]
+        else:
+            scan_steppy = [self.scan_step[1] for w in w_ar]
+
+        # Put together scan_steppx and scan_steppy
+        scan_stepp = [
+            [scan_steppx[i], scan_steppy[i]] for i in range(w_ar.shape[0])
+        ]
+        del scan_steppx
+        del scan_steppy
+
+        # Compute pos for all width
+        pos = []
+        for ch in range(len(hist)):
+            posx = [
+                np.arange(0, ref[ch].shape[0] - w[0] + 1, scan_stepp[i][0])
+                for i, w in enumerate(w_ar)
+            ]
+            posy = [
+                np.arange(0, ref[ch].shape[1] - w[1] + 1, scan_stepp[i][1])
+                for i, w in enumerate(w_ar)
+            ]
+            pos.append([
+                np.array([
+                    [p[0], p[1]]
+                    for p in itertools.product(posx[i], posy[i])
+                ])
+                for i in range(w_ar.shape[0])
+            ])
+        del posx# pos = np.array([[p[0], p[1]] for p in itertools.product(posx, posy)])
+        del posy
+
+        # Initialize p-value container for all channels, width and pos
+        res_all = np.empty((len(hist), w_ar.shape[0]), dtype=object)
```

Comment:
Why don't we just make a normal list? numpy arrays with objects are almost always not the way to go

Reply:
It is true that numpy arrays of objects don't offer much.
The reason I use numpy array here was to have a container of fixed shape, which is not the case of lists.